l10n_th_tin_service/models/res_partner.py

```diff
@@ -0,0 +1,196 @@
+        result = cl.service.ServiceTIN("anonymous", "anonymous", tin)
+        res_ord_dict = helpers.serialize_object(result)
+        _logger.log(DEBUG, pprint.pformat(res_ord_dict))
+        return res_ord_dict["vIsExist"] is not None
+
+    @staticmethod
+    def get_info_rd_vat_service(tin, branch=0):
+        """Return ordered dict with necessary result from
+           Revenue Department's web service.
+           :param tin: a string for TIN or PIN
+           :param branch: one digit of branch number
+        """
+        branch = int(branch)
+        sess = Session()
+        mod_dir = path.dirname(path.realpath(__file__))
+        cert_path = str(Path(mod_dir).parents[0]) + "/static/cert/adhq1_ADHQ5.cer"
+        sess.verify = cert_path
+        transp = Transport(session=sess)
+        try:
+            cl = Client(ResPartner.vat_web_service_url, transport=transp)
+        except requests.exceptions.SSLError:
+            _logger.log(INFO, "Fall back to unverifed HTTPS request.")
+            sess.verify = False
+            transp = Transport(session=sess)
+            cl = Client(ResPartner.vat_web_service_url, transport=transp)
+        result = cl.service.Service(
+            "anonymous",
+            "anonymous",
+            TIN=tin,
+            ProvinceCode=0,
+            BranchNumber=branch,
+            AmphurCode=0,
+        )
+        odata = helpers.serialize_object(result)
+        _logger.log(DEBUG, pprint.pformat(odata))
+        data = OrderedDict()
+        if odata["vmsgerr"] is None:
+            for key, value in odata.items():
+                if (
+                    value is None
+                    or value["anyType"][0] == "-"
+                    or key in {"vNID", "vBusinessFirstDate"}
+                ):
+                    continue
+                data[key] = value["anyType"][0]
+            return data
+        return False
+
+    @api.onchange("vat", "branch")
+    def _onchange_vat_branch(self):
+        word_map = {
+            "vBuildingName": "อาคาร ",
+            "vFloorNumber": "ชั้นที่ ",
+            "vVillageName": "หมู่บ้าน ",
+            "vRoomNumber": "ห้องเลขที่ ",
+            "vHouseNumber": "เลขที่ ",
+            "vMooNumber": "หมู่ที่ ",
+            "vSoiName": "ซอย ",
+            "vStreetName": "ถนน ",
+            "vThambol": "ต.",
+            "vAmphur": "อ.",
+            "vProvince": "จ.",
+            "vPostCode": "",
+        }
+        street_map = [
+            [
+                "vBuildingName",
+                "vRoomNumber",
+                "vFloorNumber",
+                "vHouseNumber",
+                "vStreetName",
+                "vSoiName",
+            ],
+            ["vVillageName", "vMooNumber", "vThambol"],
+        ]
+        check_branch = re.compile(r"^\d{5}$")
+        if self.branch is False:
+            self.branch = "00000"
+
+        if self.vat:
+            if ResPartner.check_rd_tin_service(self.vat):
```

Comment:
Please use this pattern to improve code readability.
`if ResPartner.check_rd_tin_service(self.vat):`
> `    ... (remaining code of the method)`

`else:`
> `    return { ... }`

to
`if not ResPartner.check_rd_tin_service(self.vat):`
> `    return { ... }`
`    `

`... (remaining code of the method)`

```suggestion
            if not ResPartner.check_rd_tin_service(self.vat):
                return {"warning": {"title": _("Validation failed."), "message": _("Failed to verify TIN or PIN %s." % self.vat), },}
                
            # remove an indentation (4 spaces) from the remaining code 
```